src/js/control-bar/time-controls/current-time-display.js

```diff
@@ -39,19 +43,35 @@ class CurrentTimeDisplay extends Component {
     });
 
     this.contentEl_ = Dom.createEl('div', {
-      className: 'vjs-current-time-display',
-      // label the current time for screen reader users
-      innerHTML: '<span class="vjs-control-text">Current Time </span>' + '0:00'
+      className: 'vjs-current-time-display'
     }, {
       // tell screen readers not to automatically read the time as it changes
       'aria-live': 'off'
-    });
+    }, Dom.createEl('span', {
+      className: 'vjs-control-text',
+      textContent: this.localize('Current Time')
+    }));
 
+    this.updateTextNode_();
     el.appendChild(this.contentEl_);
     return el;
   }
 
   /**
+   * Updates the "current time" text node with new content using the
+   * contents of the `formattedTime_` property.
+   *
+   * @private
+   */
+  updateTextNode_() {
+    if (this.textNode_) {
+      this.contentEl_.removeChild(this.textNode_);
+    }
+    this.textNode_ = document.createTextNode(` ${this.formattedTime_}`);
```

Comment:
`updateTextNode_` gets called from `createEl`, which in turn, gets called during component init which happens before we set `this.formattedTime_`. We should either have a default value for when `this.formattedTime_` isn't set, or we need to call `updateTextNode_` in the end of the constructor.
This applies to the currentTimeDisplay and duration display and whatever else uses this `updateTextNode_`.
![undefined gif](https://user-images.githubusercontent.com/535884/27566273-8b7dc94e-5b27-11e7-9aff-f89348bd70d3.gif)